### PR TITLE
feat: persona settings

### PR DESCRIPTION
**Pull request by nuanyang233**
## Description

Closes #MF-4234 MF-4533

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/mask/src/extension/popups/regexp.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export const passwordRegexp = /^(?=.{8,20}$)(?=.*[A-Z])(?=.*[a-z])(?=.*\d)(?=.*[^\dA-Za-z]).*/
```
Comment (guanbinrui) [inline packages/mask/src/extension/popups/regexp.ts]: Move to constants.

File: packages/mask/src/extension/popups/modals/SwitchPersonaModal/index.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+}))
+
+export const SwitchPersonaModal = memo<ActionModalBaseProps>(function SwitchPersonaModal({ ...rest }) {
+    const { t } = useI18N()
+    const navigate = useNavigate()
+    const { classes } = useStyles()
+    const { personas, currentPersona } = PersonaContext.useContainer()
+
+    const handleOpenDashboard = useCallback((route: DashboardRoutes) => {
+        browser.tabs.create({
```
Comment (guanbinrui) [inline packages/mask/src/extension/popups/modals/SwitchPersonaModal/index.tsx]:
```suggestion
        await browser.tabs.create({
```

File: packages/mask/src/extension/popups/modals/SwitchPersonaModal/index.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+            active: true,
+            url: browser.runtime.getURL(`/dashboard.html#${route}`),
+        })
+        if (navigator.userAgent.includes('Firefox')) {
+            window.close()
+        }
+        Services.Helper.removePopupWindow()
```
Comment (guanbinrui) [inline packages/mask/src/extension/popups/modals/SwitchPersonaModal/index.tsx]:
```suggestion
        await Services.Helper.removePopupWindow()
```

File: packages/mask/src/extension/popups/modals/SwitchPersonaModal/index.tsx [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    }, [])
+
+    const handleSelect = useCallback((persona: PersonaInformation) => {
+        Services.Settings.setCurrentPersonaIdentifier(persona.identifier).then(() => navigate(-1))
```
Comment (guanbinrui) [inline packages/mask/src/extension/popups/modals/SwitchPersonaModal/index.tsx]:
```suggestion
        await Services.Settings.setCurrentPersonaIdentifier(persona.identifier)
        navigate(-1)
```

File: packages/mask/src/extension/popups/hook/useUserContext.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        email: localStorage.getItem('email'),
+        phone: localStorage.getItem('phone'),
+        backupMethod: localStorage.getItem('backupMethod'),
+        backupAt: localStorage.getItem('backupAt'),
+    })
+
+    const updateUser = useCallback((obj: Partial<User>) => {
+        setUser((old) => {
+            const updated = { ...old, ...obj }
+            localStorage.setItem('backupPassword', btoa(updated.backupPassword ?? ''))
```
Comment (guanbinrui) [inline packages/mask/src/extension/popups/hook/useUserContext.ts:25]: Please use a single object record for storage to avoid name conflict.

Comment (nuanyang233) [reply packages/mask/src/extension/popups/hook/useUserContext.ts:25]: That may require backward compatibility with older versions.

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**